### PR TITLE
Experimental bootstrap integration

### PR DESCRIPTION
**Pull request by szhajdu**
I did a small experiment with bootstrap. Unfortunately I'm not a designer, but you can use if you like. Future improvement could be to create a one click application and rewrite frontend and backend properly. Code is small, so it shouldn't be a big deal now, but later... :)

**Changes:**
- Integrated Bootstrap components
- Integrated ChartJS to feedback history
- Integrated flatpickr for data range selection in feedback history
- Client side validation on form inputs
- Added unit tests

**Screenshot:**

home
team
team_edit
form_error
form_update
feedback_history

### CODE REVIEW

**Comment by meetmatt** [comment]
Thank you Sabee! I will give it a try tomorrow, merge and release :)
It’s probably about time to add some CI/CD love to this project to ease awesome contributions like this. Will investigate some possibilities :)

File: template/history.html.twig [hunk 1]
```diff
@@ -1,30 +1,197 @@
...
-        <tbody>
-        {% for feedback in feedbacks %}
-            <tr>
-                <td>{{ feedback.date }}</td>
-                <td>{{ feedback.rating }}</td>
-                <td>{{ feedback.comment }}</td>
-            </tr>
+    <nav aria-label="breadcrumb">
+        <ol class="breadcrumb">
+            <li class="breadcrumb-item"><a href="{{ path_for('show_team', {"id":team.id}) }}"><< Back</a></li>
```
Comment (meetmatt) [inline template/history.html.twig]: Unescaped `<` HTML entities.
`<li class="breadcrumb-item"><a href="{{ path_for('show_team', {"id":team.id}) }}">&lt;&lt; Back</a></li>`

File: template/index.html.twig [hunk 2]
```diff
@@ -1,22 +1,78 @@
...
-            </label>
-            <input class="pure-button pure-button-primary" type="submit">
-        </form>
-    </div>
-{% endblock %}
\ No newline at end of file
+{% endblock %}
+
+{% block script %}
+    <script>
```
Comment (meetmatt) [inline template/index.html.twig]: I suppose that you've duplicated this script block from the form template by accident

File: template/layout.html.twig [hunk 3]
```diff
@@ -2,22 +2,59 @@
 <html lang="en">
 <head>
     <meta charset="UTF-8">
+    <meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
     <title>Mood - Colla.io</title>
-    <link rel="stylesheet" href="https://unpkg.com/purecss@1.0.0/build/pure-min.css"
-          integrity="sha384-nn4HPE8lTHyVtfCBi5yW9d20FjT8BJwUXyWZT9InLYax14RDjBj46LmSztkmNP9w" crossorigin="anonymous">
-    <script src="https://cdn.jsdelivr.net/npm/jquery@3.4.1/dist/jquery.min.js"
```
Comment (meetmatt) [inline template/layout.html.twig]: Kudos for getting rid of this evil 👍

File: template/layout.html.twig [hunk 4]
```diff
@@ -2,22 +2,59 @@
...
+                        event.stopPropagation();
+                    }
+                    form.classList.add('was-validated');
+                }, false);
+            });
+        }, false);
+    })();
+</script>
+
+<script src="https://code.jquery.com/jquery-3.2.1.slim.min.js"
```
Comment (meetmatt) [inline template/layout.html.twig]: I think it's not required any more after you've refactored the form logic to vanilla js?

File: template/team.html.twig [hunk 5]
```diff
@@ -1,39 +1,38 @@
...
-        <form style="display:none"
-              id="form"
-              method="post"
-              class="pure-form pure-form-stacked"
-              action="{{ path_for('save_team', {'id':team.id}) }}">
-
-            <textarea style="width:50%; height: 300px" name="emails">{% for email in emails.all %}{{ email }}
+    <nav aria-label="breadcrumb">
+        <ol class="breadcrumb">
+            <li class="breadcrumb-item"><a href="{{ path_for('index') }}"><< Back</a></li>
```
Comment (meetmatt) [inline template/team.html.twig]: Unescaped entity, `&lt;&lt;`

**Review by meetmatt — CHANGES_REQUESTED** [review]
Overall it's awesome, just several tiny comments :)

**Comment by meetmatt** [comment]
Btw, this PR will also resolve 3 and 4 in the TODO section of README.md:
3. Charts.
4. Better UI.

File: template/layout.html.twig (hunk 4 above)
Comment (szhajdu) [reply template/layout.html.twig]: I would be more than happy to get rid of it, but unfortunately bootstrap still requires at least the slim version.

File: template/team.html.twig (hunk 5 above)
Comment (szhajdu) [reply template/team.html.twig]: Thx, fixed.

File: template/index.html.twig (hunk 2 above)
Comment (szhajdu) [reply template/index.html.twig]: Thx, removed.

File: template/history.html.twig (hunk 1 above)
Comment (szhajdu) [reply template/history.html.twig]: Thx, fixed.

**Pull request merged**